src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java

```diff
@@ -215,4 +215,54 @@ public int getIntValueForArgument(@NotNull final String argumentName, final int
             return defaultValue;
         }
     }
+
+    @Override
+    public int hashCode()
+    {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((actionMenu == null) ? 0 : actionMenu.hashCode());
+        result = prime * result + argumentStateByActionArgumentNameMap.hashCode();
+        return result;
+    }
+
+    @Override
+    public boolean equals(final Object obj)
+    {
+        if (this == obj)
+        {
+            return true;
+        }
+        if (obj == null)
+        {
+            return false;
+        }
+        if (getClass() != obj.getClass())
+        {
+            return false;
+        }
+        final ActionMenuState other = (ActionMenuState) obj;
+        if (actionMenu == null)
+        {
+            if (other.actionMenu != null)
+            {
+                return false;
+            }
+        }
+        else if (!actionMenu.equals(other.actionMenu))
+        {
+            return false;
+        }
+        if (!argumentStateByActionArgumentNameMap.equals(other.argumentStateByActionArgumentNameMap))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)